apps/api/src/course/course.service.ts

```diff
@@ -51,40 +58,20 @@ export class CourseService {
     return courseNos
   }
 
-  async search(
-    {
-      keyword = '',
-      genEdTypes = [],
-      dayOfWeeks = [],
-      limit = 10,
-      offset = 0,
-      periodRange,
-    }: FilterInput,
-    { semester, academicYear, studyProgram }: CourseGroupInput
-  ): Promise<Course[]> {
-    const query = {
-      semester,
-      academicYear,
-      studyProgram,
-    } as FilterQuery<CourseDocument>
-    const escapedKeyword = escapeRegExpString(keyword.trim())
-    if (keyword) {
-      query.$or = [
-        { courseNo: new RegExp('^' + escapedKeyword, 'i') },
-        { abbrName: new RegExp(escapedKeyword, 'i') },
-        { courseNameTh: new RegExp(escapedKeyword, 'i') },
-        { courseNameEn: new RegExp(escapedKeyword, 'i') },
-      ]
-    }
-
-    if (genEdTypes.length > 0) {
-      query.genEdType = { $in: genEdTypes }
-    }
-
-    if (dayOfWeeks.length > 0) {
-      query['sections.classes.dayOfWeek'] = { $in: dayOfWeeks }
-    }
-
+  async search({
```

Comment:
why even have this method in CourseService lol. Move everything to SearchService, and have CourseResolver call SearchService.